apps/src/lib/ui/SchoolInfoConfirmationDialog.jsx

```diff
@@ -15,80 +15,102 @@ export const styles = {
 class SchoolInfoConfirmationDialog extends Component {
   static propTypes = {
     schoolName: PropTypes.string,
-    onUpdate: PropTypes.func,
-    onConfirm: PropTypes.func,
-    isOpen: PropTypes.bool.isRequired
+    scriptData: PropTypes.shape({
+      formUrl: PropTypes.string.isRequired,
+      authTokenName: PropTypes.string.isRequired,
+      authTokenValue: PropTypes.string.isRequired,
+      existingSchoolInfo: PropTypes.shape({
+        id: PropTypes.number,
+        school_id: PropTypes.string,
+        country: PropTypes.string,
+        school_type: PropTypes.string,
+        school_name: PropTypes.string,
+        full_address: PropTypes.string
+      }).isRequired
+    }).isRequired,
+    onClose: PropTypes.func,
+    isOpen: PropTypes.bool
   };
 
   constructor(props) {
     super(props);
     this.state = {
       showSchoolInterstitial: false,
-      schoolName: props.schoolName,
-      isOpen: true
+      schoolName: props.scriptData.existingSchoolInfo.school_name,
+      isOpen: props.isOpen || true
     };
   }
 
-  static defaultProps = {
-    onUpdate: () => {},
-    schoolName: ''
+  handleClickYes = () => {
+    const {authTokenName, authTokenValue} = this.props.scriptData;
+    fetch(
+      `/api/v1/user_school_infos/${
+        this.props.scriptData.existingSchoolInfo.id
+      }/update_last_confirmation_date`,
+      {method: 'PATCH', headers: {[authTokenName]: authTokenValue}}
+    )
+      .then(() => this.props.onClose())
+      .catch(error => {
+        this.setState({error});
+      });
   };
 
-  componentDidMount() {
-    const {schoolName} = this.state;
-    if (!schoolName && schoolName.length > 0) {
-      fetch('/dashboardapi/v1/users/me/school_name')
-        .then(response => response.json())
-        .then(data => {
-          this.setState({
-            schoolName: data.school_name
-          });
-        })
-        .catch(error => this.setState({error}));
-    }
-  }
-
-  handleUpdateClick = () => {
+  handleClickUpdate = () => {
     this.setState({showSchoolInterstitial: true});
   };
 
-  renderInitialContent() {
-    const {onConfirm} = this.props;
+  handleClickSave = async () => {
+    const {authTokenName, authTokenValue} = this.props.scriptData;
+    fetch(
+      `/api/v1/user_school_infos/${
+        this.props.scriptData.existingSchoolInfo.id
+      }/update_end_date`,
+      {method: 'PATCH', headers: {[authTokenName]: authTokenValue}}
+    )
+      .then(() => {
+        fetch(
+          `/api/v1/users/${
+            this.props.scriptData.existingSchoolInfo.id
+          }/update_school_info_id`,
+          {method: 'PATCH', headers: {[authTokenName]: authTokenValue}}
+        ).then(() => this.props.onClose());
```

Comment:
I like that you've moved both of these changes to the final save event for the dialog.  It's left the code with this pattern:

```js
fetch('...').then(() => fetch('...'))
```

...where we will always make these two api calls one after the other.  In cases like this, it's nice to combine the two API calls into one, for a few reasons.

- Better, faster experience for the user, because they only wait for one request/response instead of two.
- Fewer failure modes to consider.  With two separate API calls, we have two failure paths we need to handle - and it's possible having the first call succeed and the second one fail will leave our saved data in an odd state.

Reply:
+1 that if we expect two API calls to be called together if possible. I definitely don't think this blocks this PR though.